blocks/api/registration.js

```diff
@@ -43,6 +43,13 @@ export function registerBlock( slug, settings ) {
 		);
 		return;
 	}
+	if ( true !== validateBlockSettings( settings ) ) {
+		console.error(
+			'Block not registered.'
+		);
+		// Return the block even though it was not registered.
+		return Object.assign( { slug }, settings );
```

Comment:
Yup, I think the second one looks better to me.  Something I like to do in my apps is have a processing chain of sorts that has consistent input output behavior.  It works similar to `Promise`s.

This could just become a pipe ( flow in lowdash ) of `blockSettings` -> `isValid` -> `addBlock` -> return settings.  That way if done properly you can interchange the processing for whatever needs may come and in the future stick something new in there pretty painlessly.